### DISCUSSION TITLE
Use '@override' decorator for methods overriden by subclass

### DISCUSSION THREAD

**Discussion by gcattan** [mobxjs/mobx#3451] [Q&A] [answered]
Hello, 

I updated a project from mobx 4 to mobx 6.6.1 and I encountered the error:

`[MobX] Cannot apply '@observable' to ...:
The field is already decorated with '@observable'.
Re-decorating fields is not allowed.
Use '@override' decorator for methods overriden by subclass.`

The same error also appear with the the decorator '@action'. 

The project is quiet big and there is a lot of places where I need to handle the suggested change.

Is there a convenient way to avoid this refactoring, and have a consistent behavior with mobx 4?

Thanks :)

**Comment by kubk** [comment]
Hi, as far as I know there is no way to automate this

**Comment by kubk** [comment, marked as answer]
TypeScript has a setting to force adding `override` TS keyword to child's methods: https://dev.to/lioness100/introducing-typescript-override-keyword-4b36

It is not the same as adding Mobx `override` keyword, but it might be helpful to run it once to detect what methods require adding `@override` Mobx decorator

**Reply by gcattan** [reply]
Hm. OK. I will try to do this, but we have around 2-3 thousand of methods with implicit override and almost the same number of reference to @action and @observable. 
Is there a lower version of mobx 6 where this behavior does not exists and that I can use in case?

**Reply by urugator** [reply]
> Is there a lower version of mobx 6 where this behavior does not exists

< 6.1, but especially inheritance was quite broken (therefore [the rework](https://github.com/mobxjs/mobx/releases/tag/mobx%406.1.0)) 

> what methods require adding @override Mobx decorator

Just to clarify a bit, `@override` is not (at least at the moment) required, leaving overriden field in subclass non-decorated is enough.

**Reply by gcattan** [reply]
Hi, 
Thanks for the explanation and sorry for the delay in my response. It took some time, but I proceeded with the typescript settings as suggested to replace all the references. 
Because the project is big, I left the `@observable` in place and use `makeObservable(this)` in the store as indicated in the doc. However, the render method calls exceed the react limit. It look like I am missing an additional setting, but I cannot find where.
